### PR TITLE
fix Announcements

### PR DESCRIPTION
**Pull request by shaubby**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x ] I have tested my code.
- [ x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ x] I have linked this PR to any issues that it closes.

Task 1: Swap the order of the announcements and Active problems/modules sections on /dashboard

Task 2: Redirect the home page to dashboard if the user is signed in. Reference this PR: https://github.com/cpinitiative/usaco-guide/pull/3689/files

Task 3: Make the text color and background color of the welcome back card the same as the active problems/modules cards

### CODE REVIEW

**Review by mjh316 — APPROVED** [review]
LGTM, ty! @SansPapyrus683 @ryanchou-dev You guys good with how it looks and affects guide? Also, any thoughts on whether we should add an "escape hatch" for those logged but still want to access the home page? like vercel.com/home iirc

**Comment by shaubby** [comment]
> LGTM, ty! @SansPapyrus683 @ryanchou-dev You guys good with how it looks and affects guide? Also, any thoughts on whether we should add an "escape hatch" for those logged but still want to access the home page? like vercel.com/home iirc

im thinking maybe we should make it so if you click on the usaco guide icon from the dashboard it doesnt automatically redirect so people can view the landing page

**Comment by mjh316** [comment]
I think that's a good idea - so in routing if like previous page was /dashboard we don't redirect, because most people coming to home from dashboard probably want to go to home. @shaubby Do you think you could impl this?

File: src/pages/index.tsx [hunk 1]
```diff
@@ -46,6 +50,13 @@ const linkTextStyles =
   'text-blue-600 dark:text-blue-300 transition hover:text-purple-600 dark:hover:text-purple-300';
 
 export default function IndexPage(): JSX.Element {
+  const firebaseUser = useFirebaseUser();
+  const loading = useIsUserDataLoaded();
+  React.useEffect(() => {
+    if (firebaseUser) {
```
Comment (mjh316) [inline src/pages/index.tsx:56]: As mentioned, add some check for if the last page (page that the user navigated from) is NOT /dashboard, and then if so, we navigate.

**Review by mjh316 — CHANGES_REQUESTED** [review]

**Comment by shaubby** [comment]
> I think that's a good idea - so in routing if like previous page was /dashboard we don't redirect, because most people coming to home from dashboard probably want to go to home. @shaubby Do you think you could impl this?

yeah i think i could

**Review by SansPapyrus683 — APPROVED** [review]
very poggers pr

**Comment by shaubby** [comment]
@mjh316 hows this

File: src/pages/dashboard.tsx [hunk 2]
```diff
@@ -168,7 +168,7 @@ export default function DashboardPage(props: PageProps) {
       <SEO title="Dashboard" />
 
       <div className="min-h-screen bg-gray-100 dark:bg-dark-surface">
-        <TopNavigationBar linkLogoToIndex={true} />
+        <TopNavigationBar linkLogoToIndex={true} dashboard={true} />
```
[the remark this answers is not in the document]
Comment (mjh316) [reply src/pages/dashboard.tsx:171]: @shaubby Whoops, I don't know if it pinged but here you go

**Review by mjh316 — CHANGES_REQUESTED** [review]

**Comment by shaubby** [comment]
Yeah im not gonna lie, it was late at night when i wrote this and i have no clue how and why it worked, but it should make sense not.
The dashboard prop is passed to the topnavigationbar to let it know that we are on the dashboard, and if you click on the usaco guide icon in the corner, it will pass the dashboard information to the location.state.redirect. Then, on the dashboard, if location.state.redirect = false, it will not automativally redirect to the dashboard. The try catch if in case the user goes straight to the landing page without coming from another page on usaco guide, location.state.redirect will not exist, and it will throw an TypeError: Cannot read properties of null exception, the try catch counteracts that.

@mjh316

**Comment by mjh316** [comment]
@shaubby That makes sense, could you quickly leave some documentation for the relevant props (maybe change dashboard to "noRedirectToDashboard" or something in case we want to reuse?) and then ping me to approve? Thank you!

**Comment by shaubby** [comment]
> @shaubby That makes sense, could you quickly leave some documentation for the relevant props (maybe change dashboard to "noRedirectToDashboard" or something in case we want to reuse?) and then ping me to approve? Thank you!

Where do you want me to leave documentation,
I went ahead and renamed it @mjh316

**Comment by mjh316** [comment]
@shaubby In the useEffect (e.g.) why try catch

**Comment by shaubby** [comment]
@mjh316 I mentioned this in my other comment, but it is in case the user opens a link straight to the landing page, and does not come from clicking the icon in the corner of the usaco guide, trying to access location.state.redirect will give the error TypeError: Cannot read properties of null exception, the try catch defaults to redirecting the user if location.state.redirect is undefined.

**Comment by mjh316** [comment]
@shaubby To clarify, I understand that you mentioned it in your other comment, I meant that it'd be great if you could document the code with comments in the useEffect and try catch area.

**Comment by ryanchou-dev** [comment]
> LGTM, ty! @SansPapyrus683 @ryanchou-dev You guys good with how it looks and affects guide? Also, any thoughts on whether we should add an "escape hatch" for those logged but still want to access the home page? like vercel.com/home iirc

i believe you can toggle between the screens logged in by clicking the usaco.guide icon

**Comment by shaubby** [comment]
Oh ok, I didnt get what you meant the first time,
I added the comments, is this ok?
@mjh316

**Review by mjh316 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
:skull:

**Pull request merged**